recipes/ldns/all/conanfile.py

```diff
@@ -23,53 +29,57 @@ class LdnsConan(ConanFile):
-        return "source_subfolder"
-
-    def validate(self):
-        if self.settings.os == "Windows":
-            raise ConanInvalidConfiguration("Windows is not supported by the ldns recipe. Contributions are welcome.")
-
     @property
     def _settings_build(self):
         return getattr(self, "settings_build", self.settings)
 
-    def requirements(self):
-        self.requires("openssl/1.1.1o")
+    def export_sources(self):
+        export_conandata_patches(self)
 
-    def build_requirements(self):
-        if self._settings_build.os == "Windows" and not tools.get_env("CONAN_BASH_PATH"):
-            self.build_requires("msys2/cci.latest")
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.rm_safe("fPIC")
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
 
-    def config_options(self):
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("openssl/1.1.1t")
+
+    def validate(self):
         if self.settings.os == "Windows":
-            del self.options.fPIC
+            raise ConanInvalidConfiguration("Windows is not supported by the ldns recipe. Contributions are welcome.")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
-
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
-
-        def yes_no(v): return "yes" if v else "no"
-        args = [
-            # libraries
-            f"--enable-shared={yes_no(self.options.shared)}",
-            f"--enable-static={yes_no(not self.options.shared)}",
-            f"--with-pic={yes_no(self.settings.os != 'Windows' and (self.options.shared or self.options.fPIC))}",
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        if not cross_building(self):
+            env = VirtualRunEnv(self)
+            env.generate(scope="build")
+
+        tc = AutotoolsDeps(self)
+        try:
+            # This fixes an issue of linking against ldns in combination of openssl:shared=False, ldns:shared=True, and an older GCC:
+            # > hidden symbol `pthread_atfork' in /usr/lib/x86_64-linux-gnu/libpthread_nonshared.a(pthread_atfork.oS) is referenced by DSO
+            # OpenSSL adds -lpthread to link with POSIX thread library. Instead, it should use -pthread compiler flag which additionally
+            # sets required macros at compile time.
+            tc.environment.remove("LIBS", "-lpthread")
+            tc.environment.append("CFLAGS", "-pthread")
+        except ValueError:
+            pass
```

Comment:
when does it raise and why?

Reply:
It raises when `-lpthread` is not in `LIBS` and it happens in some environments with static build of OpenSSL.